modules/cas_cache/main.c

```diff
@@ -42,10 +43,52 @@ MODULE_PARM_DESC(seq_cut_off_mb,
 ocf_ctx_t cas_ctx;
 struct cas_module cas_module;
 
+static inline uint32_t involuntary_preemption_enabled(void)
+{
+	bool config_dynamic = IS_ENABLED(CONFIG_PREEMPT_DYNAMIC);
+	bool config_rt = IS_ENABLED(CONFIG_PREEMPT_RT);
+	bool config_preempt = IS_ENABLED(CONFIG_PREEMPT);
+	bool config_lazy = IS_ENABLED(CONFIG_PREEMPT_LAZY);
+	bool config_none = IS_ENABLED(CONFIG_PREEMPT_NONE);
+
+	if (!config_dynamic && !config_rt && !config_preempt && !config_lazy)
+		return false;
+
+	if (config_none)
+		return false;
+
+	if (config_rt || config_preempt || config_lazy) {
+		printk(KERN_ERR OCF_PREFIX_SHORT
+				"OpenCAS doesn't support kernels with "
+				"involuntary preemption\n");
+
+		return true;
+	}
+
+#ifdef CONFIG_PREEMPT_DYNAMIC
+	/* preempt_model_none() or preempt_model_voluntary() are not defined if
+	 * the kernel has been compiled without PREEMPT_DYNAMIC
+	 */
+	printk(KERN_ERR OCF_PREFIX_SHORT
```

Comment:
Why do we print this message at the `KERN_ERR` verbosity even if the preemption is set to voluntary?